### PR TITLE
[SPARK-36853][BUILD] Code failing on checkstyle

### PR DESCRIPTION
**Pull request by Shockang**
What changes were proposed in this pull request?

Fixed the checkstyle error thrown in windows in maven installing phase.

Why are the changes needed?

Due to some problems in CI system, these obvious errors have not been detected. I think these checkstyle errors should be solved.

Does this PR introduce any user-facing change?

No.

How was this patch tested?

I have verified by executing following commands in the win10 system, except for (modifier) RedundantModifier in TimSort.java and (naming) MethodName in GroupStateTimeout.java and OutputMode.java and Trigger.java, the rest of the errors have been fixed.
`mvn -DskipTest clean install`

### CODE REVIEW

**Comment by Shockang** [comment]
@srowen Sorry, the previous PR has been inconvenient to modify because it took too long, so the code has been migrated to this PR.

**Comment by HyukjinKwon** [comment]
@Shockang the test in the GitHub Actions fails because your fork is based on the old master branch. do you mind rebasing it to sync to the latest `master` branch?

**Comment by HyukjinKwon** [comment]
ok to test

**Comment by Shockang** [comment]
> @Shockang the test in the GitHub Actions fails because your fork is based on the old master branch. do you mind rebasing it to sync to the latest `master` branch?

Of course not. I will merge the latest master code.

File: core/src/main/java/org/apache/spark/util/collection/TimSort.java [hunk 1]
```diff
@@ -120,8 +120,9 @@ public void sort(Buffer a, int lo, int hi, Comparator<? super K> c) {
     assert c != null;
 
     int nRemaining  = hi - lo;
-    if (nRemaining < 2)
+    if (nRemaining < 2) {
```
Comment (srowen) [inline core/src/main/java/org/apache/spark/util/collection/TimSort.java:123]: I think we didn't change this file as it's copied as-is from another source. Is there a way to just disable style checks? we could change the file as I doubt we'll ever need to port updates, but who knows

File: sql/hive/src/test/java/org/apache/spark/sql/hive/test/Complex.java [hunk 2]
```diff
@@ -40,15 +40,29 @@
...
-
-  private static final org.apache.thrift.protocol.TField AINT_FIELD_DESC = new org.apache.thrift.protocol.TField("aint", org.apache.thrift.protocol.TType.I32, (short)1);
-  private static final org.apache.thrift.protocol.TField A_STRING_FIELD_DESC = new org.apache.thrift.protocol.TField("aString", org.apache.thrift.protocol.TType.STRING, (short)2);
-  private static final org.apache.thrift.protocol.TField LINT_FIELD_DESC = new org.apache.thrift.protocol.TField("lint", org.apache.thrift.protocol.TType.LIST, (short)3);
-  private static final org.apache.thrift.protocol.TField L_STRING_FIELD_DESC = new org.apache.thrift.protocol.TField("lString", org.apache.thrift.protocol.TType.LIST, (short)4);
-  private static final org.apache.thrift.protocol.TField LINT_STRING_FIELD_DESC = new org.apache.thrift.protocol.TField("lintString", org.apache.thrift.protocol.TType.LIST, (short)5);
-  private static final org.apache.thrift.protocol.TField M_STRING_STRING_FIELD_DESC = new org.apache.thrift.protocol.TField("mStringString", org.apache.thrift.protocol.TType.MAP, (short)6);
+public class Complex implements org.apache.thrift.TBase<Complex, Complex._Fields>,
```
Comment (srowen) [inline sql/hive/src/test/java/org/apache/spark/sql/hive/test/Complex.java:43]: Same deal, this is copied from Hive, so maybe easier to leave it and disable checkstyle

File: core/src/main/java/org/apache/spark/util/collection/TimSort.java (hunk 1 above)
Comment (Shockang) [reply core/src/main/java/org/apache/spark/util/collection/TimSort.java:123]: > I think we didn't change this file as it's copied as-is from another source. Is there a way to just disable style checks? we could change the file as I doubt we'll ever need to port updates, but who knows

The build errors are all from the java file, the file: scalastyle-config.xml should not work. Do you know which file about java checkstyle works?

Comment (srowen) [reply core/src/main/java/org/apache/spark/util/collection/TimSort.java:123]: Look at dev/checkstyle.xml , but you probably just want to suppress them in dev/checkstyle-suppressions.xml

**Comment by srowen** [comment]
Ping @Shockang - see comments

File: sql/core/src/main/java/org/apache/spark/sql/api/java/UDF11.java [hunk 3]
```diff
@@ -26,5 +26,6 @@
  */
 @Stable
 public interface UDF11<T1, T2, T3, T4, T5, T6, T7, T8, T9, T10, T11, R> extends Serializable {
-  R call(T1 t1, T2 t2, T3 t3, T4 t4, T5 t5, T6 t6, T7 t7, T8 t8, T9 t9, T10 t10, T11 t11) throws Exception;
+  R call(T1 t1, T2 t2, T3 t3, T4 t4, T5 t5, T6 t6, T7 t7, T8 t8, T9 t9, T10 t10, T11 t11)
```
Comment (srowen) [inline sql/core/src/main/java/org/apache/spark/sql/api/java/UDF11.java:29]: This indent seems too deep - 4 spaces?

**Review by srowen — CHANGES_REQUESTED** [review]
Can we just disable checkstyle for imported Hive code though?

**Comment by srowen** [comment]
Ping @Shockang

**Pull request closed without merging**